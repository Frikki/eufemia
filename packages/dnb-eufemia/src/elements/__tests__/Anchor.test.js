/**
 * Element Test
 *
 */

import React from 'react'
import {
  mount,
  fakeProps,
  toJson,
  axeComponent
} from '../../core/jest/jestSetup'
import Component from '../Anchor'

const props = fakeProps(require.resolve('../Anchor'), {
  optional: true
})
props.element = 'a'

describe('Anchor element', () => {
  it('have to match default Anchor element snapshot', () => {
    const Comp = mount(<Component {...props}>text</Component>)
    expect(toJson(Comp)).toMatchSnapshot()
  })
  it('has dnb-a class', () => {
    const Comp = mount(<Component>text</Component>)
    expect(Comp.find('.dnb-a').exists()).toBe(true)
  })
  it('has href', () => {
    const Comp = mount(<Component href="/url">text</Component>)
    expect(Comp.find('[href]').exists()).toBe(true)
  })
  it('has no-icon class when element was given', () => {
    const Comp = mount(
      <Component href="/url" target="_blank">
        <span>text</span>
      </Component>
    )
    expect(Comp.find('.dnb-anchor--no-icon').exists()).toBe(true)
  })
  it.skip('has title when target is blank', () => {
    const Comp = mount(
      <Component href="/url" target="_blank">
        text
      </Component>
    )
    expect(Comp.find('[title]').exists()).toBe(true)
  })
  it.skip('has title about external url and target is blank', () => {
    const Comp = mount(
      <Component href="/url" target="_blank" title="External site">
        text
      </Component>
    )
    expect(
      Comp.find('[title]').get(1).instance().getAttribute('title')
    ).toBe('External site')
  })
  it('should validate with ARIA rules as a Anchor element', async () => {
    const Comp = mount(<Component {...props} />)
    expect(await axeComponent(Comp)).toHaveNoViolations()
  })
})
