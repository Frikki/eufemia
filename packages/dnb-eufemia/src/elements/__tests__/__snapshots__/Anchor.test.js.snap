// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Anchor element have to match default Anchor element snapshot 1`] = `
<ForwardRef
  className="className"
  element="a"
  href="href"
  inner_ref={null}
  omitClass={true}
  target="target"
  target_blank_title="target_blank_title"
  tooltip="tooltip"
>
  <Anchor
    className="className"
    element="a"
    href="href"
    inner_ref={null}
    omitClass={true}
    target="target"
    target_blank_title="target_blank_title"
    tooltip="tooltip"
  >
    <ForwardRef
      class={null}
      className="className"
      css={null}
      element="a"
      href="href"
      inner_ref={
        Object {
          "current": <a
            class="className dnb-a"
            element="a"
            href="href"
            target="target"
          >
            text
          </a>,
        }
      }
      internalClass={null}
      is="a"
      skeleton={null}
      skeleton_method="font"
      target="target"
      target_blank_title="target_blank_title"
    >
      <Elem
        class={null}
        className="className"
        css={null}
        element="a"
        href="href"
        inner_ref={
          Object {
            "current": <a
              class="className dnb-a"
              element="a"
              href="href"
              target="target"
            >
              text
            </a>,
          }
        }
        internalClass={null}
        is="a"
        skeleton={null}
        skeleton_method="font"
        target="target"
        target_blank_title="target_blank_title"
      >
        <a
          className="className dnb-a"
          element="a"
          href="href"
          target="target"
        >
          text
        </a>
      </Elem>
    </ForwardRef>
  </Anchor>
</ForwardRef>
`;
